nav2_route/include/nav2_route/plugins/edge_cost_functions/start_pose_orientation_scorer.hpp

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) 2024, Polymath Robotics Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef NAV2_ROUTE__PLUGINS__EDGE_COST_FUNCTIONS__START_POSE_ORIENTATION_SCORER_HPP_
+#define NAV2_ROUTE__PLUGINS__EDGE_COST_FUNCTIONS__START_POSE_ORIENTATION_SCORER_HPP_
+
+#include <memory>
+#include <string>
+
+#include "rclcpp/rclcpp.hpp"
+#include "rclcpp_lifecycle/lifecycle_node.hpp"
+#include "nav2_core/route_exceptions.hpp"
+#include "nav2_route/interfaces/edge_cost_function.hpp"
+#include "nav2_util/line_iterator.hpp"
+#include "nav2_util/node_utils.hpp"
+#include "nav2_util/robot_utils.hpp"
+#include "nav2_costmap_2d/costmap_subscriber.hpp"
+#include "tf2_geometry_msgs/tf2_geometry_msgs.hpp"
+#include "tf2/utils.h"
+#include "angles/angles.h"
+
+namespace nav2_route
+{
+
+/**
+ * @class StartPoseOrientationScorer
+ * @brief Scores initial edge edge by comparing the orientation of the robot's current pose and the orientation of the edge
```

Comment:
```suggestion
 * @brief Scores initial edge by comparing the orientation of the robot's current pose and the orientation of the edge
```

Please also provide additional information about how it scores. I don't think it actually scores as much as it rejects if > than a tolerance. (also the goal orientation one needs to be populated as well https://github.com/ros-navigation/navigation2/blob/d02bb473a743c3a44cb9291d541bb862d9232bca/nav2_route/include/nav2_route/plugins/edge_cost_functions/goal_orientation_scorer.hpp#L36)

Reply:
^ still open

Reply:
added docstring, better describing the edge scorers in https://github.com/ros-navigation/navigation2/pull/4950/commits/4124841db6d217e66195b101018c3459f41f621c